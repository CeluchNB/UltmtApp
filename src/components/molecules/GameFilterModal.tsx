import BaseModal from '../atoms/BaseModal'
import PrimaryButton from '../atoms/PrimaryButton'
import { RadioButton } from 'react-native-paper'
import React from 'react'
import SecondaryButton from '../atoms/SecondaryButton'
import TextDateInput from '../atoms/TextDateInput'
import { useTheme } from '../../hooks'
import { Controller, useForm } from 'react-hook-form'
import { StyleSheet, Text, View } from 'react-native'

interface GameFilterModalProps {
    visible: boolean
    defaultValues: { live: string; after: Date; before: Date }
    onClose: (data: any) => void
}

const GameFilterModal: React.FC<GameFilterModalProps> = ({
    visible,
    defaultValues,
    onClose,
}) => {
    const {
        theme: { colors, size, weight },
    } = useTheme()
    const { control, handleSubmit, reset } = useForm({
        defaultValues,
    })

    const styles = StyleSheet.create({
        modal: {},
        titleContainer: {
            display: 'flex',
            flexDirection: 'row',
        },
        title: {
            color: colors.textPrimary,
            fontSize: size.fontThirty,
            fontWeight: weight.bold,
            flex: 1,
        },
        text: {
            color: colors.textPrimary,
            fontSize: size.fontTwenty,
            fontWeight: weight.bold,
        },
        radioLabel: { width: 250 },
        toText: {
            margin: 20,
        },
        container: {
            display: 'flex',
            flexDirection: 'row',
            marginTop: 10,
        },
        modalContainer: {
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            display: 'flex',
            flexDirection: 'row',
            backgroundColor: `${colors.primary}50`,
        },
        modalView: {
            backgroundColor: colors.primary,
            borderColor: colors.textPrimary,
            color: colors.textPrimary,
            borderRadius: 10,
            padding: 25,
            alignItems: 'center',
            shadowColor: colors.textPrimary,
            elevation: 5,
            flex: 1,
        },
    })

    return (
        <BaseModal
            onClose={() => {
                onClose({ ...defaultValues })
            }}
            visible={visible}>
            <View style={styles.titleContainer}>
                <Text style={styles.title}>Filters</Text>
                <SecondaryButton
                    text="clear"
                    onPress={async () => {
                        reset()
                    }}
                />
            </View>
            <View>
                <View style={styles.container}>
                    <Controller
                        control={control}
                        name="live"
                        render={({ field: { value, onChange } }) => (
                            <RadioButton.Group
                                onValueChange={onChange}
                                value={value}>
                                <View style={styles.container}>
                                    <Text
                                        style={{
                                            ...styles.text,
                                            ...styles.radioLabel,
                                        }}>
                                        All Games
                                    </Text>
                                    <RadioButton
                                        value="undefined"
                                        color={colors.textPrimary}
                                        uncheckedColor={colors.textSecondary}
                                    />
                                </View>
                                <View style={styles.container}>
                                    <Text
                                        style={{
                                            ...styles.text,
                                            ...styles.radioLabel,
                                        }}>
                                        Live Games
                                    </Text>
                                    <RadioButton
                                        value="true"
                                        color={colors.textPrimary}
                                        uncheckedColor={colors.textSecondary}
                                    />
                                </View>
                                <View style={styles.container}>
                                    <Text
                                        style={{
                                            ...styles.text,
                                            ...styles.radioLabel,
                                        }}>
                                        Completed Games
                                    </Text>
                                    <RadioButton
                                        value="false"
                                        color={colors.textPrimary}
                                        uncheckedColor={colors.textSecondary}
                                    />
                                </View>
                            </RadioButton.Group>
                        )}
                    />
                </View>
                <View style={styles.container}>
                    <Controller
                        control={control}
                        name="after"
                        render={({ field: { value, onChange } }) => (
                            <TextDateInput
                                value={value}
                                onChange={onChange}
                                style={{ backgroundColor: colors.darkPrimary }}
                            />
                        )}
                    />
                    <Text style={[styles.text, styles.toText]}>to</Text>
                    <Controller
                        control={control}
                        name="before"
                        render={({ field: { value, onChange } }) => (
                            <TextDateInput
                                value={value}
                                onChange={onChange}
                                style={{ backgroundColor: colors.darkPrimary }}
                            />
                        )}
                    />
                </View>
            </View>
            <PrimaryButton
                text="done"
                onPress={async () => {
                    handleSubmit(onClose)()
                }}
                loading={false}
            />
        </BaseModal>
    )
}

export default GameFilterModal
