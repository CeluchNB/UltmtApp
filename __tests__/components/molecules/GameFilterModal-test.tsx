import GameFilterModal from '../../../src/components/molecules/GameFilterModal'
import React from 'react'
import { act, fireEvent, render } from '@testing-library/react-native'

import MockDate from 'mockdate'
MockDate.set('01 October 2022 00:00 UTC')

beforeAll(() => {
    jest.useFakeTimers({ legacyFakeTimers: true })
})

afterAll(() => {
    jest.useRealTimers()
})

it('should match snapshot', () => {
    const snapshot = render(
        <GameFilterModal
            visible={true}
            defaultValues={{
                live: 'true',
                after: new Date(),
                before: new Date(),
                showUsers: false,
                showTeams: false,
            }}
            onClose={() => {}}
        />,
    )

    expect(snapshot).toMatchSnapshot()
})

it('should call on close with correct data', async () => {
    const onClose = jest.fn()
    const { getByText } = render(
        <GameFilterModal
            visible={true}
            defaultValues={{
                live: 'true',
                after: new Date('2022-01-01'),
                before: new Date('2022-06-01'),
                showUsers: false,
                showTeams: false,
            }}
            onClose={onClose}
        />,
    )

    const doneBtn = getByText('done')
    fireEvent.press(doneBtn)
    await act(async () => {})

    expect(onClose).toBeCalled()
})

it('should close on request close', () => {
    const onClose = jest.fn()
    const { getByTestId } = render(
        <GameFilterModal
            visible={true}
            defaultValues={{
                live: 'true',
                after: new Date('2022-01-02'),
                before: new Date('2022-06-01'),
                showUsers: false,
                showTeams: false,
            }}
            onClose={onClose}
        />,
    )

    const modal = getByTestId('base-modal')
    fireEvent(modal, 'onRequestClose')

    expect(onClose).toHaveBeenCalled()
})
