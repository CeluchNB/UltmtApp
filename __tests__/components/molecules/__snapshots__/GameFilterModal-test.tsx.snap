// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  testID="base-modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#30303050",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1a1a1a",
          "borderColor": "#ffd600",
          "borderRadius": 10,
          "color": "#ffd600",
          "elevation": 5,
          "flex": 1,
          "margin": 20,
          "padding": 25,
          "shadowColor": "#ffd600",
        }
      }
    >
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "flex": 1,
              "fontSize": 61.64383561643835,
              "fontWeight": "bold",
            }
          }
        >
          Filters
        </Text>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "#ffd600",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flex": undefined,
                "minWidth": 64,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="button-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 20,
                  },
                ]
              }
              testID="button"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          {
                            "textTransform": "uppercase",
                          },
                          {
                            "color": "#ffd600",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                  testID="button-text"
                >
                  clear
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            accessibilityRole="radiogroup"
          >
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffd600",
                    "fontSize": 41.0958904109589,
                    "fontWeight": "bold",
                    "width": 250,
                  }
                }
              >
                All Games
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": false,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                color="#ffd600"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 18,
                      "padding": 6,
                    },
                  ]
                }
                uncheckedColor="#7AC6FF"
              >
                <View
                  style={
                    {
                      "opacity": 0,
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#ffd600",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ó°„¬
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffd600",
                    "fontSize": 41.0958904109589,
                    "fontWeight": "bold",
                    "width": 250,
                  }
                }
              >
                Live Games
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": true,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                color="#ffd600"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 18,
                      "padding": 6,
                    },
                  ]
                }
                uncheckedColor="#7AC6FF"
              >
                <View
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#ffd600",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ó°„¬
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffd600",
                    "fontSize": 41.0958904109589,
                    "fontWeight": "bold",
                    "width": 250,
                  }
                }
              >
                Completed Games
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": false,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                color="#ffd600"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 18,
                      "padding": 6,
                    },
                  ]
                }
                uncheckedColor="#7AC6FF"
              >
                <View
                  style={
                    {
                      "opacity": 0,
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#ffd600",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ó°„¬
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "color": "#ffd600",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    [
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                      },
                      {
                        "backgroundColor": "#1a1a1a",
                        "borderColor": "#7AC6FF",
                        "borderRadius": 4,
                        "borderWidth": 1,
                      },
                      undefined,
                    ]
                  }
                  testID="text-input-outline"
                />
                <View>
                  <View
                    style={
                      [
                        {
                          "paddingBottom": 0,
                        },
                        {
                          "minHeight": 56,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <TextInput
                      cursorColor="#ffd600"
                      editable={true}
                      focusable={false}
                      maxFontSizeMultiplier={1.5}
                      multiline={false}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      onPressOut={[Function]}
                      placeholder="10/01/22"
                      placeholderTextColor="#ffd600"
                      selectionColor="#ffd600"
                      style={
                        [
                          {
                            "margin": 0,
                            "zIndex": 1,
                          },
                          {
                            "height": 48,
                          },
                          {
                            "paddingBottom": 0,
                            "paddingTop": 0,
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "letterSpacing": 0.15,
                            "lineHeight": undefined,
                            "paddingHorizontal": 16,
                            "textAlign": "left",
                            "textAlignVertical": "center",
                          },
                          false,
                          [
                            {},
                          ],
                          undefined,
                        ]
                      }
                      testID="date-text-input"
                      underlineColorAndroid="transparent"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Text
            style={
              [
                {
                  "color": "#ffd600",
                  "fontSize": 41.0958904109589,
                  "fontWeight": "bold",
                },
                {
                  "margin": 20,
                },
              ]
            }
          >
            to
          </Text>
          <View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "color": "#ffd600",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    [
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                      },
                      {
                        "backgroundColor": "#1a1a1a",
                        "borderColor": "#7AC6FF",
                        "borderRadius": 4,
                        "borderWidth": 1,
                      },
                      undefined,
                    ]
                  }
                  testID="text-input-outline"
                />
                <View>
                  <View
                    style={
                      [
                        {
                          "paddingBottom": 0,
                        },
                        {
                          "minHeight": 56,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <TextInput
                      cursorColor="#ffd600"
                      editable={true}
                      focusable={false}
                      maxFontSizeMultiplier={1.5}
                      multiline={false}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      onPressOut={[Function]}
                      placeholder="10/01/22"
                      placeholderTextColor="#ffd600"
                      selectionColor="#ffd600"
                      style={
                        [
                          {
                            "margin": 0,
                            "zIndex": 1,
                          },
                          {
                            "height": 48,
                          },
                          {
                            "paddingBottom": 0,
                            "paddingTop": 0,
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "letterSpacing": 0.15,
                            "lineHeight": undefined,
                            "paddingHorizontal": 16,
                            "textAlign": "left",
                            "textAlignVertical": "center",
                          },
                          false,
                          [
                            {},
                          ],
                          undefined,
                        ]
                      }
                      testID="date-text-input"
                      underlineColorAndroid="transparent"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#ffd600",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "#ffd600",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": "auto",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 24,
                          "marginVertical": 10,
                        },
                        {
                          "marginHorizontal": 8,
                        },
                        {
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                    ],
                  ]
                }
                testID="button-text"
              >
                done
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
